docs/howto/how-to-make-api-tool/api-tool-guide.md

```diff
@@ -0,0 +1,250 @@
+
+```python
+    class APITool(PluginV2):
+        """Concrete implementation of an AyxPlugin."""
+
+        def __init__(self, provider: AMPProviderV2) -> None:
+            self.provider = provider
+            # truncated code
+
+        def on_incoming_connection_complete(self, anchor: namedtuple) -> None:
+            # truncated code
+
+        def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+            # truncated code
+
+        def on_complete(self) -> None:
+            import pandas as pd
+            import pyarrow as pa
+
+            df = pd.DataFrame(
+                {
+                    "x": [1, 2, 3],
+                    "y": ["hello", "world", "from ayx_python_sdk!"],
+                    "z": [self.config_value, self.config_value, self.config_value],
+                }
+            )
+
+            packet = pa.Table.from_pandas(df)
+
+            self.provider.write_to_anchor("Output", packet)
+            self.provider.io.info("APITool tool done.")
+```
+---
+## Writing our API request tool
+---
+Now you are ready to begin modifying this plugin code to pull data from an API and tell the plugin to output it! We will use the [requests](https://requests.readthedocs.io/en/latest/) library to do this.
+
+> Since this is an input tool, we will only focus on the `on_complete` function. For additional reading on the lifecycle of a plugin, refer to the [Ayx Python SDK documentation](https://alteryx.github.io/ayx-python-sdk/plugin_lifecycle.html)
+
+### 1. Making the request
+For this example, we will be fetching data from the [BALLDONTLIE API](https://app.balldontlie.io/) for NBA player Lebron James' stats in the 2016 NBA Playoffs. 
+
+First, we want to get rid of the existing boilerplate code in the `on_complete` function, and leave only the `import pyarrow as pa` line. Your code should now look like this:
+
+```python
+    def on_complete(self) -> None:
+        import pyarrow as pa
+        import pyarrow.compute as pc # add this line in, it will be used later
+```
+
+Next, we will write a function to make a GET request to the API, passing in Lebron James' `player_id` and `season`:
+```python
+    def get_postseason_stats(player_id=237, season=2016):
+        resp = requests.get(
+            "https://www.balldontlie.io/api/v1/stats",
+            params={
+                "player_ids[]": player_id,
+                "postseason": True,
+                "seasons[]": season,
+            },
+        )
+        return resp.json()["data"]
+```
+From calling this function with `player_id=237` and `season=2016`, you will get back the data as a JSON.
+After returning the JSON data, we're able to read that into a [`pyarrow.Table`](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html)
+
+### 2. Creating a `pyarrow.Table`
+Since we read and write data in the [Apache Arrow](https://arrow.apache.org/) data format, we will now convert this JSON to `pyarrow.Table` format with the following function:
```

Comment:
TW review - update to:

Since we read and write data in the [Apache Arrow](https://arrow.apache.org/) data format, we need to convert this JSON to `pyarrow.Table` format with this function: